content/source/assets/javascripts/application.js

```diff
@@ -7,9 +7,45 @@
 
 //= require analytics
 
+// In navigation sidebars, hide most sub-lists and reveal any that contain the
+// current page. The a.current-page class is added during build by
+// layouts/inner.erb.
 document.addEventListener("turbolinks:load", function() {
     "use strict";
     var docsSidebar = $("#docs-sidebar ul.nav.docs-sidenav");
     docsSidebar.find("ul.nav").addClass("nav-hidden");
     docsSidebar.find("li").has("a.current-page").addClass("active");
 });
+
+// On docs/content pages, add a hierarchical quick nav menu if there are any
+// H2/H3/H4 headers.
+document.addEventListener("turbolinks:load", function() {
+    "use strict";
```

Comment:
any reason why this is only in this function rather than at the top level?

Reply:
- I thought it was only function scope? 
- If not, would it mess with anything being pulled in by the `//= require` statements?

Anyway, I'll defer to you!

Reply:
It can be global as well, but that's a fair question about messing with other statements. Honestly I don't think you have any code here that violates strict mode anyway so it probably makes little difference, but usually i see people who use it putting it right at the top of files.